lib/puppet/node/environment.rb

```diff
@@ -271,10 +271,11 @@ def modules
       seen_modules = {}
       modulepath.each do |path|
         Dir.entries(path).each do |name|
-          next if name == "." || name == ".."
+          next unless Puppet::Module.is_module?(name, path)
```

Comment:
I think that is not the problem trying to be solved by PUP-3678, but should probably go into a new ticket.